### ISSUE TITLE
Feature request: newline-before-return rule for .jsx components

### ISSUE THREAD

**Issue by 1alexvash** [jsx-eslint/eslint-plugin-react#3532]
I would like a similar rule to this one [this one](https://eslint.org/docs/latest/rules/newline-before-return)? But for **JSX** components only

Example of incorrect code ❌

```ts
const Component = () => {
  // Some statefulll logic...
  return <div/>;
};
```

Example of correct code ✔️

```ts
const Component = () => {
  // Some statefulll logic...

  return <div/>;
};
```

This would rule would help distingish the **logic** part of the component from the UI part
